packages/client/hmi-client/src/services/goLLM.ts

```diff
@@ -100,47 +98,15 @@ export async function compareModels(
 	modelIds: string[],
 	workflowId?: string,
 	nodeId?: string
-): Promise<CompareModelsResponseType> {
-	let resolve;
-	let reject;
-
-	const promise = new Promise<CompareModelsResponseType>((res, rej) => {
-		resolve = res;
-		reject = rej;
+): Promise<TaskResponse> {
+	const { data } = await API.get<TaskResponse>('/gollm/compare-models', {
+		params: {
+			'model-ids': modelIds.join(','),
+			'workflow-id': workflowId,
+			'node-id': nodeId
+		}
 	});
-
-	try {
-		const response = await API.get<TaskResponse>('/gollm/compare-models', {
-			params: {
-				'model-ids': modelIds.join(','),
-				'workflow-id': workflowId,
-				'node-id': nodeId
-			}
-		});
-
-		const taskId = response.data.id;
-
-		await handleTaskById(taskId, {
-			ondata(data, closeConnection) {
-				if (data?.status === TaskStatus.Failed) {
-					closeConnection();
-					logger.warn(`Failed to Compare Models with ids: ${modelIds.join(',')}`);
-					reject({ response: '' });
-				}
-				if (data.status === TaskStatus.Success) {
-					closeConnection();
-
-					// data.output is a base64 encoded json object. We decode it and return the json object.
-					const str = b64DecodeUnicode(data.output);
-					resolve(JSON.parse(str));
```

Comment:
Why removing it from here and add it to the component itself?

Reply:
Because the gollm request is async and it just returns the status object without the output and the event listener for listening Success event is now registered on the component.